enhancements/microshift/microshift-route-admission-policy.md

```diff
@@ -0,0 +1,146 @@
+---
+title: microshift-route-admission-policy
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@eslutsky"
+  - "@copejon"
+  - "@ggiguash"
+  - "@pmtk"
+  - "@pliurh"
+  - "@jerpeter1"
+  - "@Miciah"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2024-01-30
+last-updated: 2024-02-01
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-1067
+see-also:
+  - "/enhancements/ingress/openshift-route-admission-policy.md"
+---
+
+# MicroShift route admission policy
+## Summary
+OpenShift defaults to not allow routes in multiple namespaces use the same
+hostname, and MicroShift inherits that default.
+
+Ever since OpenShift 4 this has been possible to configure, and MicroShift
+should allow that too to accommodate additional use cases.
+
+## Motivation
+Users who compose applications from multiple sources may deploy those to
+different namespaces, albeit want to expose them using the same hostname and
+different path.
+
+### User Stories
+As a MicroShift admin, I want to configure MicroShift to support combining
+routes from multiple namespaces.
+
+### Goals
+- Allow the use of the same host in routes from different namespaces.
+
+### Non-Goals
+N/A
+
+## Proposal
+The proposal follows all the changes that were done in the original
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/openshift-route-admission-policy.md).
+
+This includes the exposure of a new configuration option to pass on to the
+router to allow cross-namespace host routes.
+
+MicroShift will default to disable the namespace ownership checks, contrary to
+OpenShift. MicroShift is not meant to be multi-tenant and hold several
+unrelated applications that need protection between them.
+
+MicroShift is also focused on single-node deployments, and runs without an
+external load balancer and in environments without access to deep DNS
+integration. The primary means of accessing the apps on the MicroShift host
```

Comment:
Is there a secondary means?

Reply:
You could have a few of them, but depends on the application. Additional routers, different ingress controllers, direct service exposure, etc.
All of those are out of control for MicroShift, the default router is provided as a helper and a means of easily accessing applications.